highlighting/lib/src/highlightv2.dart

```diff
@@ -122,29 +123,28 @@ class HighlightV2 {
     }
 
     void processSubLanguage() {
-      if (modeBuffer == '') {
+      assert(
+        top.subLanguage != null,
+        'processSublanguage can only be called with non-null sublanguage',
+      );
+
+      if (modeBuffer.isEmpty) {
         return;
       }
-      Result result;
-      // TODO(yescorp): Find out how to handle multiple sublanguages
-      if ((top.subLanguage?.length ?? 0) >= 1) {
-        if (builtinLanguages[top.subLanguage!.first] == null) {
-          emitter.addText(modeBuffer);
-          return;
-        }
 
-        // check type of subLanguage
+      Result result;
+      if (top.subLanguage?.length == 1) {
         result = highlight(top.subLanguage!.first, modeBuffer, true,
             continuation: continuations[top.subLanguage!.first]);
         continuations[top.subLanguage!.first] = result.top;
       } else {
-        result = Result(language: 'unknown');
+        result = highlightAuto(modeBuffer, top.subLanguage!);
       }
 
-      if (result.relevance != null && top.relevance! > 0) {
-        relevance += result.relevance!;
+      if (top.relevance! > 0) {
```

Comment:
Seems like we can't make relevance non-nullable.
Because 0 means 0, and null means unspecified.
While compiling a language definition, we set null relevance to be 1.